src/sparseml/optim/helpers.py

```diff
@@ -347,7 +347,9 @@ def check_if_staged_recipe(container: dict) -> bool:
     """
     for k, v in container.items():
         if isinstance(v, dict):
-            if any([key for key in v.keys() if "modifiers" in key]):
+            if any(
+                [key for key in v.keys() if isinstance(key, str) and "modifiers" in key]
```

Comment:
nit:can be absolutely ignored, but would be good to take advantage of `any(...)` method's early stopping by using generator

```suggestion
                key for key in v.keys() if isinstance(key, str) and "modifiers" in key
```